model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/remotestore/RemoteCacheSessionListener.java

```diff
@@ -160,14 +183,22 @@ protected void replaceRemoteEntityInCache(K key, long eventVersion) {
 
             SessionEntityWrapper<V> sessionWrapper = remoteSession.mergeRemoteEntityWithLocalEntity(localEntityWrapper);
 
-            // We received event from remoteCache, so we won't update it back
-            replaced = cache.getAdvancedCache().withFlags(Flag.SKIP_CACHE_STORE, Flag.SKIP_CACHE_LOAD, Flag.IGNORE_RETURN_VALUES)
-                    .replace(key, localEntityWrapper, sessionWrapper);
+            KeycloakModelUtils.runJobInTransaction(sessionFactory, (session -> {
+
+                RealmModel realm = session.realms().getRealm(sessionWrapper.getEntity().getRealmId());
+                long lifespanMs = lifespanMsLoader.apply(realm, sessionWrapper.getEntity());
+                long maxIdleTimeMs = maxIdleTimeMsLoader.apply(realm, sessionWrapper.getEntity());
```

Comment:
Could this be extracted out of the `do`-`while` loop? Session creation is costly and it is unlikely that `lifespanMs` and `maxIdleTimeMs` would change during the loop.

Reply:
The `KeycloakModelUtils.runJobInTransaction` can be seen inside the `do`-`while` loop. Could it be extracted completely?

Reply:
Replied to this in the other comment https://github.com/keycloak/keycloak/pull/7736#discussion_r569302965 .